compiler/src/dotty/tools/dotc/config/Settings.scala

```diff
@@ -281,49 +319,64 @@ object Settings:
       setting
     }
 
-    def BooleanSetting(name: String, descr: String, initialValue: Boolean = false, aliases: List[String] = Nil): Setting[Boolean] =
-      publish(Setting(name, descr, initialValue, aliases = aliases))
+    @unshared
+    val settingCharacters = "[a-zA-Z0-9_\\-]*".r
+    def validateSetting(setting: String): String =
+      assert(settingCharacters.matches(setting), s"Setting $setting contains invalid characters")
+      setting 
+
+    def validateAndPrependName(name: String): String =
+      assert(!name.startsWith("-"), s"Setting $name cannot start with -")
+      "-" + validateSetting(name)
+
+    def BooleanSetting(category: String, name: String, descr: String, initialValue: Boolean = false, aliases: List[String] = Nil): Setting[Boolean] =
+      publish(Setting(category, validateAndPrependName(name), descr, initialValue, aliases = aliases.map(validateSetting)))
```

Comment:
It might be simpler to pass the `name` as is and validate it in `Setting`.

Reply:
It prepends the "-" for backward compatibility, so the names don't change. But we don't pass "-" anymore in the config. Little magic but seems to be optimal out of all posibilities.